web/src/settings_org.js

```diff
@@ -396,10 +396,21 @@ export function populate_auth_methods(auth_methods) {
     auth_methods = settings_components.sort_object_by_key(auth_methods);
     let rendered_auth_method_rows = "";
     for (const [auth_method, value] of Object.entries(auth_methods)) {
+        // Certain authentication methods are not available to be enabled without
+        // purchasing a plan, so we need to disable them in this UI.
+        // The restriction only applies to **enabling** the auth method, so this
+        // logic is dependent on the current value.
+        // The reason for that is that if for any reason, the auth method is already
+        // enabled (for example, because it was manually enabled for the organization
+        // by request, as an exception) - the organization should be able to disable it
+        // if they don't want it anymore.
+        const cant_be_enabled =
+            page_params.unavailable_authentication_methods.includes(auth_method) && !value;
+
         rendered_auth_method_rows += render_settings_admin_auth_methods_list({
             method: auth_method,
             enabled: value,
-            disable_configure_auth_method: !can_configure_auth_methods(),
+            disable_configure_auth_method: !can_configure_auth_methods() || cant_be_enabled,
```

Comment:
AzureAD and SAML are greyed out in the UI now:

![image](https://github.com/zulip/zulip/assets/45007152/88612b07-2fab-466c-921b-c506c12e7a36)

I think we may want to also add a little explanation that will pop-up when you mouse over them saying "You need to upgrade to X plan to unlock this"?

Reply:
Yeah, let's add tooltips to those options when they are disabled. Something like this should work for the wording:

> Upgrade to Zulip Cloud Plus to enable this feature.
> Upgrade to Zulip Cloud Standard or Zulip Cloud Plus to enable this feature.